src/hal/armv7/_init-imxrt.S

```diff
@@ -25,9 +25,17 @@
 .extern syscalls
 .extern syspage
 
+#define IOMUXC_GPR_GPR16 0x400ac040
+#define IOMUXC_GPR_GPR17 0x400ac044
+
+#define INTERNAL_FLASH_START 0x70000000
+
+#define SYSP_PROGS_SIZE_ADDR 0x20c
```

Comment:
No need for those defines. Use syspage label + hardcoded offsets
e.g.
ldr r1, =syspage
ldr r0, [r1, #0xc] /* size */
ldr r0, [r1, #0x10] /* progs */